src/main/java/org/scify/jedai/progressivejoin/TopkGlobal.java

```diff
@@ -54,12 +57,11 @@ else if (Math.abs((Double) o1[2]-(Double) o2[2])<1E-13 && (int)o1[0] == (int)o2[
         });
         records.forEach((recordid, tokens) -> events.add(new Object[] {recordid, 0, 1.0}));
 
-        // <simval, recordid1, recordid2>
-        SortedList<Object[]> tmpresults = new SortedList<>((o1, o2) -> {
-            if ((Double)o1[0] > (Double)o2[0])

```

Comment:
Please double-check the logic change here. I moved the `return 0` case in front of the `return -1` thinking the intent was that similarity values that were very close (within some rounding error) were to be considered equal.